api/swagger-spec/oapi-v1.json

```diff
@@ -22067,7 +22071,7 @@
      },
      "config": {
       "$ref": "v1.ObjectReference",
-      "description": "config is an ObjectReference to the BuildConfig this Build is based on."
+      "description": "config is an ObjectReference to the BuildConfig this Build is based on. This field is deprecated in deference to Spec.Config."
```

Comment:
You can't make spec.config authoritative, that is not backwards compatible for a client that doesn't know about spec.config.  

I don't see a way to add this to v1 without it being a breaking change.